packages/core/react/src/WalletProvider.tsx

```diff
@@ -124,8 +124,12 @@ export function WalletProvider({
         if (autoConnect !== true || !adapter) {
             return;
         }
-        return adapter.connect.bind(adapter);
-    }, [adapter, autoConnect]);
+        if (walletName === SolanaMobileWalletAdapterWalletName && getIsMobile(adapters)) {
+            return (adapter as SolanaMobileWalletAdapter).autoConnect_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.bind(adapter);
```

Comment:
This exists because `connect()` does not distinguish between user-initiated connection requests and auto-initiated connection requests. In MWA, if the connection requests is auto-initiated, we do _not_ want to try to open the mobile wallet app; the OS will block such an attempt unless the request was initiated by a user gesture (eg. a tap or a swipe).

This special codepath attempts to load a prior authorization out of the cache, but otherwise does not contact the wallet if one can't be found.

Reply:
While I enjoy the vibe of `autoConnect_DO_NOT_USE_OR_YOU_WILL_BE_FIRED`, does the warning really matter? This is a nonstandard method that other adapters won't have, but would it matter if a dapp wanted to call it directly?

Reply:
Also, there's something a bit weird about a method `handleAutoConnect` that... does not handle auto connect, but instead returns a bound connect function?

Reply:
> …but would it matter if a dapp wanted to call it directly?

I say yes, because I want to be able to make this method go away one day, once `connect({silent: true})` is a thing. Me adding the suffix to this – a public method – is just a way of capturing plausible deniability when I rug it on someone who decided to rely on it anyway.

> …there's something a bit weird about a method handleAutoConnect that... does not handle auto connect…

Yeah, this is me fitting a pattern where:

* The _prop_ that a component takes for an event handler must be called `onFoo`
* The _function_ that gets called when that event happens must be called `handleFoo`

From the perspective of this code:

* `onAutoConnect` is what `WalletProviderBase` calls when it decides it wants to trigger an auto-connection. This is analagous to a `<button>` firing `onChange`
* `handleAutoConnect` is the function that `WalletProvider` calls to handle that trigger.

To get really pedantic I could call them `onAutoConnectRequest` and `handleAutoConnectRequest`.

Reply:
I'll do that.